README.md

```diff
@@ -9,7 +9,11 @@ PolyKey requires a public/private keypair for all crypto operations, this can be
 The major dependencies of js-polykey are:
 - [kbpgp](https://github.com/keybase/kbpgp) for all asymmetric cryptography.
 - [EncryptedFS](https://github.com/MatrixAI/js-encryptedfs) for symmetric encryption within Vaults.
-- [isomorphi-git](https://github.com/isomorphic-git/isomorphic-git) for version control within Vaults.
+- [isomorphic-git](https://github.com/isomorphic-git/isomorphic-git) for version control within Vaults.
+
+## Class Dependencies
+In order to better understand how the pieces of PolyKey fit together, use this dependency DAG as reference:
+<img src="./media/dependencies.png">
 
 ## KeyManager
 This class is responsible for managing the public and private keys as well as any crypto operations using those keys. The symmetric vault keys are also managed by this instance.
```

Comment:
Something I always do when doing with resources like file descriptors is to always to use the bracket pattern.

This means keymanager should be able to take a file descriptor, and assume something else has created it already.

This works more nicely if the language already has a "file" object.

The closest equivalent in Nodejs ecosystem is the `ReadStream` and `WriteStream` and that should mean we need to be using the `readable-stream` library to provide a user-space streams implementation. We use this in virtualfs too.

Note that when receiving streams, it is expected that the caller owns the streams, this allows you pass the responsibility of resource management to the caller, and they can do sync/async try finally brackets.

However if you take paths, then you take the responsibility of opening and closing and resource management. Make sure to keep track of sync/async try finally brackets.

It's always a good idea to allow the caller to make the responsibility. Because this keeps the code more composable.

I'm just saying this with respect to the KeyManager, but it's probably applicable to all the other parts of js-polykey.